src/averageAges.js

```diff
@@ -54,11 +67,26 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  // 1. keep people who have mothers in the array
+  // and Each person (or only for men)
+
+  const children = people.filter((child) => people.find(mother => onlyWithSon
+    ? mother.name === child.mother && child.sex === 'm'
+    : mother.name === child.mother
+  ));
+
+  // 2. calculate the difference child.born - mother.born
+  const ageDifference = children.map(function(person) {
+    return person.born
+    - people[people.findIndex((human) => human.name === person.mother)].born;
```

Comment:
```suggestion
    - people.find((human) => human.name === person.mother).born;
```
shorter way
find theoretically can return `undefined` but in this case `findIndex` will return `-1` and `people[-1] === undefined` as well